utils/python/CIME/hist_utils.py

```diff
@@ -123,9 +124,33 @@ def _hists_match(model, hists1, hists2, suffix1="", suffix2=""):
     two_not_one = sorted([hists2[normalized2.index(item)] for item in set_of_2_not_1])
 
     both = set(normalized1) & set(normalized2)
+
     match_ups = sorted([ (hists1[normalized1.index(item)], hists2[normalized2.index(item)]) for item in both])
-    expect(len(match_ups) + len(set_of_1_not_2) == len(hists1), "Programming error1")
-    expect(len(match_ups) + len(set_of_2_not_1) == len(hists2), "Programming error2")
+    if multiinst:
```

Comment:
Can you please add some high-level comments about what the code in this conditional is doing? I *think* what this is saying is: If there's a file with a multi-instance in one set, but this doesn't exist in the other set, and the same file without a multi-instance suffix exists in the other set, then consider these two to be a match. But I was only able to figure that out based on knowledge of the issue you were fixing by adding this code.

In general, this routine has gotten complicated enough, including some complex regex matches, that it's hard to understand the big picture of what each piece of code is doing. I'd really love to see some more comments throughout this routine, and/or have it broken up into evocatively-named lower-level functions. For example, it would help if you gave an example of what normalized_name will look like, what multi_normalized will look like, and the purposes of the various pieces of this routine.